### ISSUE TITLE
[IntelliJ] Allow to import icons from web sites

### ISSUE THREAD

**Issue by egorikftp** [ComposeGears/Valkyrie#189]
Epic for IntelliJ IDEA plugin to natively import icons from different third-party icon providers.

The list of most popular services:

| Icons provider                                                               | License                                                                                                                                                                        | Status | PR                                                |
|------------------------------------------------------------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|--------|---------------------------------------------------|
| [Material Symbols](https://fonts.google.com/icons?icon.set=Material+Symbols) | [![GitHub License](https://img.shields.io/github/license/google/material-design-icons?style=flat-square)](https://github.com/google/material-design-icons/blob/master/LICENSE) | ✅      | https://github.com/ComposeGears/Valkyrie/pull/664 |
| [Lucide](https://github.com/lucide-icons/lucide)                             | [![GitHub License](https://img.shields.io/github/license/lucide-icons/lucide?style=flat-square)](https://github.com/lucide-icons/lucide/blob/main/LICENSE)                     | ✅      | https://github.com/ComposeGears/Valkyrie/pull/781 |
| [Heroicons](https://github.com/tailwindlabs/heroicons)                       | [![GitHub License](https://img.shields.io/github/license/tailwindlabs/heroicons?style=flat-square)](https://github.com/tailwindlabs/heroicons/blob/master/LICENSE)             | ❌      |                                                   |
| [Bootstrap icons](https://github.com/twbs/icons)                             | [![GitHub License](https://img.shields.io/github/license/twbs/icons?style=flat-square)](https://github.com/twbs/icons/blob/main/LICENSE)                                       | ✅      | https://github.com/ComposeGears/Valkyrie/pull/858 |
| [Simple icons](https://github.com/simple-icons/simple-icons)                 | [![GitHub License](https://img.shields.io/github/license/simple-icons/simple-icons?style=flat-square)](https://github.com/simple-icons/simple-icons/blob/develop/LICENSE.md)   | ❌      |                                                   |
| [Feather](https://github.com/feathericons/feather)                           | [![GitHub License](https://img.shields.io/github/license/feathericons/feather?style=flat-square)](https://github.com/feathericons/feather/blob/main/LICENSE)                   | ❌      |                                                   |
| [Eva Icons](https://github.com/akveo/eva-icons)                              | [![GitHub License](https://img.shields.io/github/license/akveo/eva-icons?style=flat-square)](https://github.com/akveo/eva-icons/blob/main/LICENSE)                             | ❌      |                                                   |
| [Font Awesome](https://github.com/FortAwesome/Font-Awesome)                  | [![GitHub License](https://img.shields.io/badge/license-CC_BY_4.0-green?style=flat-square)](https://github.com/FortAwesome/Font-Awesome/blob/6.x/LICENSE.txt)                  | ❌      |                                                   |
| [Octicons](https://github.com/primer/octicons)                               | [![GitHub License](https://img.shields.io/github/license/primer/octicons?style=flat-square)](https://github.com/primer/octicons/blob/main/LICENSE)                             | ❌      |                                                   |
| [css.gg](https://github.com/astrit/css.gg)                                   | [![GitHub License](https://img.shields.io/github/license/astrit/css.gg?style=flat-square)](https://github.com/astrit/css.gg/blob/main/LICENSE.md)                              | ❌      |                                                   |               
| [Tabler](https://github.com/tabler/tabler-icons)                             | [![GitHub License](https://img.shields.io/github/license/tabler/tabler-icons?style=flat-square)](https://github.com/tabler/tabler-icons/blob/main/LICENSE)                     | ❌      |                                                   |
| [Remix Icon](https://github.com/Remix-Design/RemixIcon)                      | [![GitHub License](https://img.shields.io/github/license/Remix-Design/RemixIcon?style=flat-square)](https://github.com/Remix-Design/RemixIcon/blob/master/License)             | ❌      |                                                   |
| [Boxicons](https://github.com/atisawd/boxicons)                              | [![GitHub License](https://img.shields.io/github/license/atisawd/boxicons?style=flat-square)](https://github.com/atisawd/boxicons/blob/master/LICENSE)                         | ❌      |                                                   |
| [Ionicons](https://github.com/ionic-team/ionicons)                           | [![GitHub License](https://img.shields.io/github/license/ionic-team/ionicons?style=flat-square)](https://github.com/ionic-team/ionicons/blob/main/LICENSE)                     | ❌      |                                                   |
| [svgrepo](https://www.svgrepo.com/)                                          |                                                                                                                                                                                | ❌      |                                                   |

**Comment by t-regbs** [comment]
Hello @egorikftp I want to pick this up, but looking at compose icons and I can see it consolidates a few icon packs. I'd like to suggest we prioritize direct implementations for popular icon providers rather than relying solely on Compose Icons as an intermediary.
Going direct allows us have:
- Provider-Specific Features like we have for Material Symbols(Variable fonts fill, weight, Outlined/Rounded/Sharp etc)
These features would be lost or limited through Compose Icons abstraction.
- one less abstraction layer
- No dependency on third-party update cycles
- only fetch icons when browsing that provider
- direct access to latest icons/features


Think we can do:
1: Direct Implementations 
- Material Symbols 3,476 icons - Already implemented
- Lucide 1,520 icons
- Heroicons 324 icons
- Bootstrap Icons 1,355 icons
...

2: Compose Icons Integration
As a "More Icon Packs" option for less-common providers
VSCode codicon, Radix Icons, etc.

Let me know what you think

**Comment by egorikftp** [comment]
@t-regbs Hello, I agree that a direct implementation can bring us more benefits 🙂 
I will update the epic with the actual icon providers

**Comment by egorikftp** [comment]
@t-regbs will we do smth with **svgrepo** ? Looks like it quite complex.

**Comment by t-regbs** [comment]
> [@t-regbs](https://github.com/t-regbs) will we do smth with **svgrepo** ? Looks like it quite complex.

Yeah it does look complicated and many of the icons don't look free to use. Think we can leave svgrepo out unless there's demand for it.

**Comment by egorikftp** [comment]
So, I'll close for now as completed 🥳. 
Thanks for your contribution.
